src/static/js/webmentions.js

```diff
@@ -0,0 +1,288 @@
+// Code related to parsing and showing webmentions
+
+// Gets the webmentions json data from current URL
+async function getWebmentions(targetURL) {
+  const apiURL = `https://webmention.io/api/mentions.jf2?perPage=500&target=${targetURL}`;
+  let mentions = []
+  try {
+    const response = await window.fetch(apiURL);
+    if (response.status >= 200 && response.status < 300) {
+      json = await response.json();
+      mentions = json.children;
+    } else {
+      console.error("Could not parse response", response.statusText);
+    }
+  } catch(error) {
+    console.error("Request failed", error);
+  }
+  return mentions;
+}
+
+// Format published date into human readable form
+function formatDate(dateString){
+  const options = { year: "numeric", month: "long", day: "numeric" }
+  return new Date(dateString).toLocaleDateString(undefined, options)
+}
+
+// Parse webmentions for individual category
+function parseMentions(webmentions, mentionType) {
+  let filteredMentions = []
+  webmentions.forEach(function(mention){
+    if (mention["wm-property"] == mentionType) {
+      filteredMentions.push(mention)
+    }
+  });
+  return filteredMentions;
+}
+
+// Renders webmention likes
+function renderLikes(likes) {
+  // Add the likes count to the likes-tab
+  document.querySelector('#likes-count').innerHTML = likes.length;
```

Comment:
Could we reimplement every use of `innerHTML` with something like `createElement` and `appendChild`?